api/query_engine.go

```diff
@@ -28,6 +37,11 @@ func (a archives) Len() int           { return len(a) }
 func (a archives) Swap(i, j int)      { a[i], a[j] = a[j], a[i] }
 func (a archives) Less(i, j int) bool { return a[i].interval < a[j].interval }
 
+// takes a ttl and returns the oldest timestamp with that ttl
+func oldestTs(ttl uint32) uint32 {
+	return uint32(time.Now().Unix()) - ttl
```

Comment:
I think we should use the same `time.Now()` to process everything within one alignRequests, otherwise weird things could happen if the clock progresses.